app/services/visit_request/cancel.rb

```diff
@@ -6,10 +6,23 @@ def initialize(visit_request)
 
     def call
       visit_request.canceled!
+      attend_first_user_from_waitlist
     end
 
     private
 
     attr_reader :visit_request
+
+    def attend_first_user_from_waitlist
+      return unless waiting_list_request = VisitRequest.waiting_list.first
+
+      waiting_list_request.main_list!
```

Comment:
Consider such situation:
```
VisitRequest::Approve
```
raises error, but it will not rollback `waiting_list_request` to the waiting list.
These 2 operations should be in transactions.

Please add such spec:
1. stub VisitRequest::Approve to raise an error
2. request waiting list should be rollbacked to the `false` state